astroid/inference.py

```diff
@@ -824,7 +824,7 @@ def _to_literal(node: nodes.NodeNG) -> Any:
 
 def _do_compare(
     left_iter: Iterable[nodes.NodeNG], op: str, right_iter: Iterable[nodes.NodeNG]
-) -> "bool | type[util.Uninferable]":
+) -> Union[bool, util.Uninferable]:
```

Comment:
> Ah yeah, I didn't think about the implications of this. Well, then we also don't need the ignores because since `mypy` infers `Uninferable` incorrectly the typing is actually correct in its eyes.

They are offset by 1 (i.e. from the right column to the middle one). Fortunately for us, all are so it still works (at least here).